### PR TITLE
README.md updates for the alpha

### PR DESCRIPTION
**Pull request by mikebrow**
Adds a user guide to the readme.

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,21 +5,76 @@
...
+for information about current and future milestones.
+## Social
+This is a project within the Kubernetes SIG named `sig-node`.
+For more information about `sig-node` and the `cri-containerd` project:
+* [sig-node community site](https://github.com/kubernetes/community/tree/master/sig-node).
+* Catch us on Kubernetes (kubernetes.slack.com) in the `#sig-node` channel.
+* Mailing List: https://groups.google.com/forum/#!forum/kubernetes-sig-node
+## Getting started
+### Binary Dependencies
+The alpha release of cri-containerd has been tested on [containerd](https://github.com/containerd/containerd) v0.2.3-1098-g8ed1e24, [runc](https://github.com/opencontainers/runc) v1.0.0-rc3, and current releases of [CNI](https://github.com/containernetworking/cni).
```
Comment (Random-Liu) [inline README.md]: Let's use a list:
* containerd ...
* runc ...
* cni ...

And we are using cni V0.4.0.

File: README.md [hunk 2]
```diff
@@ -5,21 +5,76 @@
...
+### Build cri-containerd
+1. Install and setup a go1.8.x development environment.
+1. Make a local clone of this repository.
+1. From your github.com/kubernetes-incubator/cri-containerd/ path (Note: run these commands as root):
+```
+make install.tools
+make
+make install
```
Comment (Random-Liu) [inline README.md]: `sudo make install` instead? It seems that only this need to be root.

File: README.md [hunk 3]
```diff
@@ -5,21 +5,76 @@
...
+* [sig-node community site](https://github.com/kubernetes/community/tree/master/sig-node).
+* Catch us on Kubernetes (kubernetes.slack.com) in the `#sig-node` channel.
+* Mailing List: https://groups.google.com/forum/#!forum/kubernetes-sig-node
+## Getting started
+### Binary Dependencies
+The alpha release of cri-containerd has been tested on [containerd](https://github.com/containerd/containerd) v0.2.3-1098-g8ed1e24, [runc](https://github.com/opencontainers/runc) v1.0.0-rc3, and current releases of [CNI](https://github.com/containernetworking/cni).
+### Build cri-containerd
+1. Install and setup a go1.8.x development environment.
+1. Make a local clone of this repository.
+1. From your github.com/kubernetes-incubator/cri-containerd/ path (Note: run these commands as root):
```
Comment (Random-Liu) [inline README.md]: Only last command require root.

Comment (Random-Liu) [inline README.md]: From `github.com/kubernetes-incubator/cri-containerd` run:
```
make
sudo make install
````

File: README.md [hunk 4]
```diff
@@ -5,21 +5,76 @@
...
+```
+make install.tools
```
Comment (Random-Liu) [inline README.md]: Seems not required for installation.

File: README.md [hunk 5]
```diff
@@ -5,21 +5,76 @@
...
+make
+make install
+```
+### Running with cri-tools
```
Comment (Random-Liu) [inline README.md]: Don't think we need to include this here. Should be in developer guide.

File: README.md [hunk 6]
```diff
@@ -5,21 +5,76 @@
...
+This is a project within the Kubernetes SIG named `sig-node`.
+For more information about `sig-node` and the `cri-containerd` project:
+* [sig-node community site](https://github.com/kubernetes/community/tree/master/sig-node).
+* Catch us on Kubernetes (kubernetes.slack.com) in the `#sig-node` channel.
+* Mailing List: https://groups.google.com/forum/#!forum/kubernetes-sig-node
+## Getting started
+### Binary Dependencies
+The alpha release of cri-containerd has been tested on [containerd](https://github.com/containerd/containerd) v0.2.3-1098-g8ed1e24, [runc](https://github.com/opencontainers/runc) v1.0.0-rc3, and current releases of [CNI](https://github.com/containernetworking/cni).
+### Build cri-containerd
+1. Install and setup a go1.8.x development environment.
```
Comment (Random-Liu) [inline README.md]: 1. XX
2. XXX
3. XXX

File: README.md [hunk 7]
```diff
@@ -5,21 +5,76 @@
...
+
+1. Start `containerd` in a first terminal
+```
+containerd -l debug
+```
+2. Start `cri-containerd` in a second terminal
+```
+cri-containerd -v 4 --alsologtostderr
+```
+3. From the kubernetes project directory startup a local cluster using cri-containerd for the runtime implementation
```
Comment (Random-Liu) [inline README.md]: `cri-containerd`.

File: README.md [hunk 8]
```diff
@@ -5,21 +5,76 @@
...
+```
+KUBERNETES_PROVIDER=local CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/cri-containerd.sock' ./hack/local-up-cluster.sh
```
Comment (Random-Liu) [inline README.md]: Is `KUBERNETES_PROVIDER=local` required?

File: README.md (hunk 6 above)
Comment (mikebrow) [inline README.md]: not need it auto numbers the list

File: README.md (hunk 3 above)
Comment (mikebrow) [inline README.md]: I can change it to "may" require root if you prefer.. but it needs root on my system

File: README.md (hunk 4 above)
Comment (mikebrow) [inline README.md]: again depends..

File: README.md (hunk 2 above)
Comment (mikebrow) [inline README.md]: again depends..

File: README.md (hunk 5 above)
Comment (mikebrow) [inline README.md]: disagree

File: README.md (hunk 8 above)
Comment (mikebrow) [inline README.md]: Unless they changed it the default is GCE not local.

File: README.md (hunk 5 above)
Comment (mikebrow) [inline README.md]: How about we change the text to reflect your new make install.deps and make test-cri.. then just point them to cri-tools cli that they installed with make install.deps (presuming you agree that install.deps should have the cli-tools installed)..

File: README.md [hunk 9]
```diff
@@ -5,21 +5,79 @@
...
+* `crictl`: a command line interface (CLI) for managing pods and containers.
+
+1. Install cri-tools
+```
+go get github.com/kubernetes-incubator/cri-tools/cmd/critest
+go get github.com/kubernetes-incubator/cri-tools/cmd/crictl
+```
+2. Run `containerd` in a first terminal
+```
+containerd -l debug
```
Comment (Random-Liu) [inline README.md]: Don't think we need debug level, user can set it themselves if they care about it.

And please use:

\`\`\`shell
$ containerd
\`\`\`

And same for the following code blocks.

File: README.md [hunk 10]
```diff
@@ -5,21 +5,79 @@
...
+```
+3. Run `cri-containerd` in a second terminal
+```
+cri-containerd -v 4 --alsologtostderr
```
Comment (Random-Liu) [inline README.md]: Let's use level 2 as default. 4 is too spammy now.

File: README.md [hunk 11]
```diff
@@ -5,21 +5,79 @@
...
+```
+4. Run cri-tool's verification tests against cri-containerd in a third terminal
+```
+critest --runtime-endpoint=/var/run/cri-containerd.sock validation
```
Comment (Random-Liu) [inline README.md]: Let's skip unsupported test.

File: README.md [hunk 12]
```diff
@@ -5,21 +5,79 @@
...
+```
+You can also test out cri-containerd via [crictl](https://github.com/kubernetes-incubator/cri-tools#crictl).
```
Comment (Random-Liu) [inline README.md]: Let's make it consistent. Either always using `cri-containerd`, or always cri-containerd.

File: README.md [hunk 13]
```diff
@@ -5,21 +5,79 @@
...
+### Running with Kubernetes
+If you already have a working development environment for Kubernetes, you can
+try cri-containerd in a local cluster:
+
+1. Start `containerd` in a first terminal
+```
+containerd -l debug
```
Comment (Random-Liu) [inline README.md]: Same, no need to set log level.

File: README.md [hunk 14]
```diff
@@ -5,21 +5,79 @@
...
+```
+2. Start `cri-containerd` in a second terminal
+```
+cri-containerd -v 4 --alsologtostderr
```
Comment (Random-Liu) [inline README.md]: 2

File: README.md (hunk 8 above)
Comment (Random-Liu) [inline README.md]: Not required. Let's keep it consistent with https://github.com/kubernetes/community/blob/master/contributors/devel/local-cluster/local.md#starting-the-cluster.

File: README.md (hunk 4 above)
Comment (Random-Liu) [inline README.md]: Why do we need `make install.tools` for compile?

File: README.md (hunk 6 above)
Comment (Random-Liu) [inline README.md]: Just nit to make the original markdown more readable. :)

File: README.md (hunk 11 above)
Comment (Random-Liu) [inline README.md]: Test to skip `Streaming|RunAsUser|host port`, and explain the reason briefly.

File: README.md [hunk 15]
```diff
@@ -5,21 +5,79 @@
...
+use with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones)
+for information about current and future milestones.
+## Social
+This is a project within the Kubernetes SIG named `sig-node`.
+For more information about `sig-node` and the `cri-containerd` project:
+* [sig-node community site](https://github.com/kubernetes/community/tree/master/sig-node).
+* Catch us on Kubernetes (kubernetes.slack.com) in the `#sig-node` channel.
+* Mailing List: https://groups.google.com/forum/#!forum/kubernetes-sig-node
+## Getting started
+### Binary Dependencies
```
Comment (Random-Liu) [inline README.md]: Let's also tell user how to install the dependencies.

Comment (Random-Liu) [inline README.md]: At least we should provide links.

File: README.md [hunk 16]
```diff
@@ -5,21 +5,79 @@
...
+The alpha release of cri-containerd has been tested on:
+* [containerd](https://github.com/containerd/containerd) v0.2.3-1098-g8ed1e24
+* [runc](https://github.com/opencontainers/runc) v1.0.0-rc3
+* [CNI](https://github.com/containernetworking/cni) v0.4.0
```
Comment (Random-Liu) [inline README.md]: Add a explanation here that we'll switch to containerd 1.0 after container releases 1.0.

File: README.md [hunk 17]
```diff
@@ -5,21 +5,79 @@
 cri-containerd is a [containerd](https://containerd.io/) based implementation of Kubernetes [container runtime interface (CRI)](https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/kubelet/api/v1alpha1/runtime/api.proto).
 ![cri-containerd](./docs/cri-containerd.png)
 ## Current Status
-We are currently working toward an alpha version to be used in conjunction with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones) for information about current and future milestones.
+The current release of `cri-containerd` is `v0.1.0-alpha.1`. This release is for
+use with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones)
+for information about current and future milestones.
+## Social
```
Comment (mikebrow) [inline README.md]: I'll move it down and pick one to make consistent with.. say /service-catalog

File: README.md (hunk 15 above)
Comment (mikebrow) [inline README.md]: I did provide links... but I'll also mention the new install.deps that we'll have to keep up to date :-)

**Comment by mikebrow** [comment]
Comments addressed.

File: README.md [hunk 18]
```diff
@@ -2,23 +2,75 @@
...
+```shell
+sudo containerd
+```
+2. Start `cri-containerd` as root in a second terminal:
+```shell
+sudo cri-containerd -v 2 --alsologtostderr
+```
+3. From the kubernetes project directory startup a local cluster using `cri-containerd`:
+```shell
+KUBERNETES_PROVIDER=local CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/cri-containerd.sock' ./hack/local-up-cluster.sh
```
Comment (Random-Liu) [inline README.md]: As is mentioned in https://github.com/kubernetes-incubator/cri-containerd/pull/89#discussion_r124410361, `KUBERNETES_PROVIDER=local` is not required.

File: README.md [hunk 19]
```diff
@@ -2,23 +2,75 @@
 [![Build Status](https://api.travis-ci.org/kubernetes-incubator/cri-containerd.svg?style=flat-square)](https://travis-ci.org/kubernetes-incubator/cri-containerd)
 [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes-incubator/cri-containerd?style=flat-square)](https://goreportcard.com/report/github.com/kubernetes-incubator/cri-containerd)
 
-cri-containerd is a [containerd](https://containerd.io/) based implementation of Kubernetes [container runtime interface (CRI)](https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/kubelet/api/v1alpha1/runtime/api.proto).
+`cri-containerd` is a [containerd](https://containerd.io/) based implementation of Kubernetes [container runtime interface (CRI)](https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/kubelet/api/v1alpha1/runtime/api.proto).
 ![cri-containerd](./docs/cri-containerd.png)
 ## Current Status
-We are currently working toward an alpha version to be used in conjunction with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones) for information about current and future milestones.
+The current release of `cri-containerd` is `v0.1.0-alpha.1`. This release is for
```
Comment (Random-Liu) [inline README.md]: cri-containerd is in Alpha. It works with Kubernetes 1.7+. See the roadmap...

Mentioned in https://github.com/kubernetes-incubator/cri-containerd/pull/89#discussion_r124402843.

File: README.md [hunk 20]
```diff
@@ -2,23 +2,75 @@
...
 * Champion: Yuju Hong ([@yujuhong](https://github.com/yujuhong))
-* SIG: [sig-node](https://github.com/kubernetes/community/tree/master/sig-node)
+* SIG: `sig-node`
 
-For more information about sig-node, please check out:
-* [community site](https://github.com/kubernetes/community/tree/master/sig-node).
-* Slack: #sig-node
+For more information about `sig-node` and the `cri-containerd` project:
+* [sig-node community site](https://github.com/kubernetes/community/tree/master/sig-node).
+* Catch us on Kubernetes (kubernetes.slack.com) in the `#sig-node` channel.
```
Comment (Random-Liu) [inline README.md]: Slack: `#sig-node` channel in Kubernetes (kubernetes.slack.com).

File: README.md [hunk 21]
```diff
@@ -2,23 +2,75 @@
...
+`cri-containerd` is a [containerd](https://containerd.io/) based implementation of Kubernetes [container runtime interface (CRI)](https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/kubelet/api/v1alpha1/runtime/api.proto).
 ![cri-containerd](./docs/cri-containerd.png)
 ## Current Status
-We are currently working toward an alpha version to be used in conjunction with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones) for information about current and future milestones.
+The current release of `cri-containerd` is `v0.1.0-alpha.1`. This release is for
+use with Kubernetes 1.7. See the [roadmap](./docs/proposal.md#roadmap-and-milestones)
+for information about current and future milestones.
+## Getting started
+### Binary Dependencies and Specifications
+The alpha release of cri-containerd has been tested with:
```
Comment (Random-Liu) [inline README.md]: The current release of `cri-containerd`.

File: README.md [hunk 22]
```diff
@@ -2,23 +2,75 @@
...
+As containerd and runc move to their respective general availability releases,
+we will do our best to rebase/retest `cri-containerd` with these releases on a
+weekly/monthly basis. Similarly, given that `cri-containerd` uses the Open
+Container Initiative (OCI) [image](https://github.com/opencontainers/image-spec)
+and [runtime](https://github.com/opencontainers/runtime-spec) specifications, we
+will also do our best to update `cri-containerd` to the latest releases of these
+specifications as appropriate.
+### Install Dependencies and Build cri-containerd
+1. Install and setup a go1.8.x development environment.
+1. Make a local clone of this repository.
```
Comment (Random-Liu) [inline README.md]: Need to install several libraries first.
We should include [this section](https://github.com/kubernetes-incubator/cri-containerd/pull/94/files#diff-5a61e9f9f0b3221e1b12f6e24cf94ed4R26) here, and I could link here in the test doc.

File: README.md [hunk 23]
```diff
@@ -2,23 +2,75 @@
...
+1. From your github.com/kubernetes-incubator/cri-containerd/ path:
+```shell
+# Note: install.deps installs the above mentioned binary dependencies and cri-tools (discussed below)
+make install.deps
+make
+sudo make install
+```
+### Running with cri-tools
```
Comment (Random-Liu) [inline README.md]: `Validate the setup` or `Validate the installation`.

File: README.md [hunk 24]
```diff
@@ -2,23 +2,75 @@
...
+Another Kubernetes incubator project called [cri-tools](https://github.com/kubernetes-incubator/cri-tools)
```
Comment (Random-Liu) [inline README.md]: [CRI validation test](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md) is a test framework validating that a Container Runtime Interface (CRI) implementation meets all the requirements.

File: README.md [hunk 25]
```diff
@@ -2,23 +2,75 @@
...
+includes programs for exercising CRI implementations such as `cri-containerd`.
+For example, cri-tools includes `critest` which is a testing tool for running
+CRI verification and benchmark tests.
+Use `critest` to validate your installation of `cri-containerd`:
```
Comment (Random-Liu) [inline README.md]: s/`critest`/CRI validation test

File: README.md (hunk 22 above)
Comment (mikebrow) [inline README.md]: you can build without installing containerd and runc.. I mention install.deps below

Comment (Random-Liu) [inline README.md]: `install.deps` also depends on those libraries. It will not install the libraries itself.

File: README.md (hunk 23 above)
Comment (mikebrow) [inline README.md]: That's what it says below where it explains validation

File: README.md (hunk 24 above)
Comment (mikebrow) [inline README.md]: That's what it says below..

File: README.md (hunk 25 above)
Comment (mikebrow) [inline README.md]: ok..

File: README.md (hunk 8 above)
Comment (mikebrow) [inline README.md]: From the link you provided:
"export KUBERNETES_PROVIDER=local"
Exporting the environment variable is the same as setting at the command line.

File: README.md (hunk 18 above)
Comment (mikebrow) [inline README.md]: see response to comment above.

**Comment by mikebrow** [comment]
Comments addressed, all but `local` comment see above explanation.

File: README.md (hunk 8 above)
Comment (Random-Liu) [inline README.md]: Now it's not the only way. When you run hack/local-cluster-up.sh, there are 2 options, you can choose either.

File: README.md (hunk 23 above)
Comment (Random-Liu) [inline README.md]: I mean as a instruction, reader doesn't care about what is `cri-tools`. It makes reader confusing "Why i need to use `cri-tools` in this step?".

The real purpose is to `validate my setup` as you mentioned in https://github.com/kubernetes-incubator/cri-containerd/pull/89#discussion_r124015426, which makes more sense to the user.

File: README.md (hunk 22 above)
Comment (mikebrow) [inline README.md]: hmm

File: README.md (hunk 23 above)
Comment (mikebrow) [inline README.md]: ok

File: README.md (hunk 8 above)
Comment (mikebrow) [inline README.md]: but it's not a developer.. and local is need to run locally.. are you saying change to show multiple options if they have a GCE account?

Comment (mikebrow) [inline README.md]: but yeah ok will remove the provider = local variable

File: README.md (hunk 22 above)
Comment (mikebrow) [inline README.md]: ok lot of text there.. but ok

**Comment by mikebrow** [comment]
Ok.. all comments addressed. I completely refactor'd ### Install Dependencies and ### Build and Install cri-containerd sections to incorporate the library dependencies so please re-read those.

File: README.md [hunk 26]
```diff
@@ -2,23 +2,91 @@
...
+## Getting started
+### Binary Dependencies and Specifications
+The current release of cri-containerd has been tested with:
+* [containerd](https://github.com/containerd/containerd) v0.2.3-1098-g8ed1e24
+* [runc](https://github.com/opencontainers/runc) v1.0.0-rc3
+* [CNI](https://github.com/containernetworking/cni) v0.4.0
+
+As containerd and runc move to their respective general availability releases,
+we will do our best to rebase/retest `cri-containerd` with these releases on a
+weekly/monthly basis. Similarly, given that `cri-containerd` uses the Open
```
Comment (Random-Liu) [inline README.md]: I think we'll be using the same OCI version with runc and containerd, right? Then, maybe don't need to mention them here or say we'll be using the same version with containerd/runc.

If containerd or runc doesn't update, we can't.

File: README.md [hunk 27]
```diff
@@ -2,23 +2,91 @@
...
+Container Initiative (OCI) [image](https://github.com/opencontainers/image-spec)
+and [runtime](https://github.com/opencontainers/runtime-spec) specifications, we
+will also do our best to update `cri-containerd` to the latest releases of these
+specifications as appropriate.
+### Install Dependencies
+1. runc dependencies:
+* Install the libsecomp development library appropriate for your distribution,
```
Comment (Random-Liu) [inline README.md]: `libseccomp-dev`(Ubuntu, Debian) / `libseccomp-devel`(Fedora, CentOS, RHEL). On
 releases of Ubuntu <=Trusty and Debian <=jessie a backport version of
 `libsecomp-dev` is required. See [travis yaml](.travis.yml) for an example on
 trusty.

I feel like it's good enough.

File: README.md [hunk 28]
```diff
@@ -2,23 +2,91 @@
...
+
+As containerd and runc move to their respective general availability releases,
+we will do our best to rebase/retest `cri-containerd` with these releases on a
+weekly/monthly basis. Similarly, given that `cri-containerd` uses the Open
+Container Initiative (OCI) [image](https://github.com/opencontainers/image-spec)
+and [runtime](https://github.com/opencontainers/runtime-spec) specifications, we
+will also do our best to update `cri-containerd` to the latest releases of these
+specifications as appropriate.
+### Install Dependencies
+1. runc dependencies:
```
Comment (Random-Liu) [inline README.md]: nit: Install runc dependencies

A step should start with a verb. (Our document guys told me)

File: README.md [hunk 29]
```diff
@@ -2,23 +2,91 @@
...
+* Install the libsecomp development library appropriate for your distribution,
+`libseccomp-dev`(Ubuntu, Debian) / `libseccomp-devel`(Fedora, CentOS, RHEL). On
+releases of Ubuntu <=Trusty and Debian <=jessie a backport version of
+`libsecomp-dev` is required. See [travis yaml](.travis.yml) for an example on
+trusty.
+2. containerd dependencies:
```
Comment (Random-Liu) [inline README.md]: nit: Install containerd dependencies.

File: README.md [hunk 30]
```diff
@@ -2,23 +2,91 @@
...
+* btrfs-tools(Ubuntu, Debian) / btrfs-progs-devel(Fedora, CentOS, RHEL)
```
Comment (Random-Liu) [inline README.md]: `btrfs-tools` to keep consistent.

File: README.md [hunk 31]
```diff
@@ -2,23 +2,91 @@
...
+3. Install and setup a go1.8.x development environment.
+4. Make a local clone of this repository.
+5. Install binary dependencies by running the following command from your github.com/kubernetes-incubator/cri-containerd/ path:
```
Comment (Random-Liu) [inline README.md]: nit: From `cri-containerd` project directory?

You use it for Kubernetes project, why not make it the same here?

File: README.md [hunk 32]
```diff
@@ -2,23 +2,91 @@
...
+```shell
+# Note: install.deps installs the above mentioned runc, containerd, and CNI
+# binary dependencies as well as cri-tools (discussed below). install.deps is
```
Comment (Random-Liu) [inline README.md]: cri-tools is not our dependency and it's not installed in `install.deps`.

File: README.md [hunk 33]
```diff
@@ -2,23 +2,91 @@
...
+make install.deps
+```
+### Build and Install cri-containerd
+To build and install cri-containerd enter the following commands from your github.com/kubernetes-incubator/cri-containerd/ path:
+```shell
+make
+sudo make install
+```
+### Validate Your cri-containerd Setup
+Another Kubernetes incubator project called [cri-tools](https://github.com/kubernetes-incubator/cri-tools)
```
Comment (Random-Liu) [inline README.md]: Still think we should emphasize CRI Validation test here, and link to their document https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md, which has more clear information than the cri-tools repo.

File: README.md [hunk 34]
```diff
@@ -2,23 +2,91 @@
...
+```shell
+# Note: install.deps installs the above mentioned runc, containerd, and CNI
+# binary dependencies as well as cri-tools (discussed below). install.deps is
+# only provided for general use and ease of testing. To customize `runc`
+# and `containerd` build tags and/or to configure `cni`, please follow
+# instructions in their documents.
+make install.deps
+```
+### Build and Install cri-containerd
+To build and install cri-containerd enter the following commands from your github.com/kubernetes-incubator/cri-containerd/ path:
```
Comment (Random-Liu) [inline README.md]: From your `cri-containerd` project directory.

File: README.md (hunk 26 above)
Comment (mikebrow) [inline README.md]: said another way when they GA we all need to upgrade.. don't think it hurts to mention OCI here...

File: README.md (hunk 28 above)
Comment (mikebrow) [inline README.md]: yeah the title says install, but ok

File: README.md (hunk 31 above)
Comment (mikebrow) [inline README.md]: good point changing to:
```
from your cloned `cri-containerd/` project directory
```

File: README.md (hunk 32 above)
Comment (mikebrow) [inline README.md]: oops :-)

File: README.md (hunk 33 above)
Comment (mikebrow) [inline README.md]: ok I'll make that emphasis

Comment (mikebrow) [inline README.md]:
```
Another Kubernetes incubator project called [cri-tools](https://github.com/kubernetes-incubator/cri-tools)
includes programs for exercising CRI implementations such as `cri-containerd`.
More importantly, cri-tools includes the program `critest` which is used to for running
[CRI Validation](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md) and benchmark tests. 
Run the CRI Validation test to validate your installation of `cri-containerd`:
...
```

**Comment by mikebrow** [comment]
nits addressed

File: README.md (hunk 27 above)
Comment (Random-Liu) [inline README.md]: Sorry, I think maybe I didn't explain it clearly here.

I mean we don't need to literally explain "Install the libsecomp development library appropriate for your distribution", just
```
1. runc dependencies:
* `libseccomp-dev` (Ubuntu, Debian) / `libseccomp-devel` (Fedora, CentOS, RHEL). On
releases of Ubuntu <=Trusty and Debian <=jessie a backport version of
`libsecomp-dev` is required. See [travis.yml](XXXX) for an example on
trusty.
```

File: README.md (hunk 30 above)
Comment (Random-Liu) [inline README.md]: Same here. The original one LGTM, we just need to put it into backquote to make it consistent with before.

File: README.md [hunk 35]
```diff
@@ -2,23 +2,87 @@
...
+5. Install binary dependencies by running the following command from your cloned `cri-containerd/` project directory:
+```shell
+# Note: install.deps installs the above mentioned runc, containerd, and CNI
+# binary dependencies. install.deps is only provided for general use and ease of
+# testing. To customize `runc` and `containerd` build tags and/or to configure
+# `cni`, please follow instructions in their documents.
+make install.deps
+```
+### Build and Install cri-containerd
+To build and install cri-containerd enter the following commands from your gcri-containerd project directory:
```
Comment (Random-Liu) [inline README.md]: get rid of "g", and add backquotes.

File: README.md (hunk 33 above)
Comment (Random-Liu) [inline README.md]: 1) CRI validation test includes the validation and benchmark.
2) Their document has already linked to cri-tools, and explained what is cri-tools. We really don't need to mention it at the beginning here.

```
[CRI validation test](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md) is a test suite validating that a Container Runtime Interface (CRI) implementation such as `cri-containerd` meets all the requirements.
```

If you want, you can then add:
```
It's part of [`cri-tools`](https://github.com/kubernetes-incubator/cri-tools) which contains a group of toolkits for CRI development.
```

File: README.md (hunk 35 above)
Comment (mikebrow) [inline README.md]: need a wysiwyg editor for md docs..

File: README.md [hunk 36]
```diff
@@ -2,23 +2,89 @@
...
+and [runtime](https://github.com/opencontainers/runtime-spec) specifications, we
+will also do our best to update `cri-containerd` to the latest releases of these
+specifications as appropriate.
+### Install Dependencies
+1. Install runc dependencies.
+* runc requires installation of the libsecomp development library appropriate for your distribution. `libseccomp-dev` (Ubuntu, Debian) / `libseccomp-devel` (Fedora, CentOS, RHEL). On releases of Ubuntu <=Trusty and Debian <=jessie a backport version of
+`libsecomp-dev` is required. See [travis.yml](XXXX) for an example on
+trusty.
+2. Install containerd dependencies.
+* containerd requires installation of `btrfs-tools`. `btrfs-tools`(Ubuntu, Debian) / `btrfs-progs-devel`(Fedora, CentOS, RHEL)
```
Comment (Random-Liu) [inline README.md]: requires installation of btrfs development library.

File: README.md [hunk 37]
```diff
@@ -2,23 +2,89 @@
...
+To build and install `cri-containerd` enter the following commands from your `cri-containerd` project directory:
+```shell
+make
+sudo make install
+```
+### Validate Your cri-containerd Setup
+Another Kubernetes incubator project called [cri-tools](https://github.com/kubernetes-incubator/cri-tools)
+includes programs for exercising CRI implementations such as `cri-containerd`.
+More importantly, cri-tools includes the program `critest` which is used for running
+[CRI Validation](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md) and benchmark tests.
```
Comment (Random-Liu) [inline README.md]: running [CRI Validation Test](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md).

Get rid of "benchmark tests". It's part of "CRI Validation Tests". We may also use `make test-cri` with appropriate arguments to run benchmark in the future.

Comment (mikebrow) [inline README.md]: gotcha

**Pull request merged**